ragChain.py

```diff
@@ -0,0 +1,129 @@
+import os
+from dotenv import load_dotenv
+import pymongo
+import openai
+import pprint
+import streamlit as st
+
+from langchain_community.document_loaders import TextLoader
+from langchain_community.document_loaders import PyPDFLoader
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.runnables import RunnablePassthrough
+from langchain_mongodb import MongoDBAtlasVectorSearch
+from langchain_openai import ChatOpenAI, OpenAIEmbeddings
+from langchain_core.prompts import ChatPromptTemplate
+from langchain.prompts import PromptTemplate
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain.docstore.document import Document
+# from langchain_community.embeddings import HuggingFaceBgeEmbeddings
+# from langchain_community.vectorstores import FAISS
+# from langchain_groq import ChatGroq
+
+load_dotenv(override=True)
+
+# Initialize OpenAI
+OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
+openai.api_key = OPENAI_API_KEY
+
+
+# MongoDB setup
+MONGO_DB_URI = str(os.getenv("MONGO_DB_URI"))
+MONGO_DB_NAME = str(os.getenv("MONGO_DB_NAME"))
+MONGO_COLLECTION_NAME = str(os.getenv("MONGO_COLLECTION_NAME"))
+
+mongo = pymongo.MongoClient(MONGO_DB_URI)
+db = mongo[MONGO_DB_NAME]
+atlas_collection = db[MONGO_COLLECTION_NAME]
+vector_search_index = "vector_index"
+
+rag_chain: any
+retriever: any
+
+def create_chunks():
+    if not st.session_state.transcript:
+            return []
+    
+    # Using CharacterTextSplitter for using transcript directly
+    # text_splitter = CharacterTextSplitter(chunk_size=200, chunk_overlap=20)
```

Comment:
we can probably just get rid of unused code at the point

Reply:
Sure, will do, removed in latest commit.
Had just found two different approaches to pass in the transcript to the splitter - 1) document and 2) text. Both required different text_splitter mechanisms.